src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V> [] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold || table == null) {
+            resize();
+        }
+        putValue(key, value);
+        size++;
+    }
 
+    private void putValue(K key, V value) {
+        int numberOfBucket = hash(key) % table.length;
+        if (table[numberOfBucket] != null) {
+            Node<K, V> currentElement = table[numberOfBucket];
+            if (Objects.equals(currentElement.key, key)) {
+                currentElement.value = value;
+                size--;
+            } else {
+                while (currentElement.next != null) {
+                    if (Objects.equals(currentElement.key, key)) {
+                        currentElement.value = value;
+                        size--;
+                        return;
+                    }
+                    currentElement = currentElement.next;
+                }
+                currentElement.next = new Node<>(key, value);
+            }
+        } else {
+            table[numberOfBucket] = new Node<>(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            int numberOfBucket = hash(key) % table.length;
+            Node<K, V> element = table[numberOfBucket];
+            if (Objects.equals(element.key, key)) {
+                return element.value;
+            } else {
+                Node<K, V> currentElement = element;
+                while (currentElement != null) {
+                    if (Objects.equals(currentElement.key, key)) {
+                        return currentElement.value;
+                    }
+                    currentElement = currentElement.next;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V> [] oldTable = table;
+        int oldCapacity = (oldTable == null) ? 0 : table.length;
+        int oldThreshold = threshold;
+        int newCapacity;
+        int newThreshold;
+        if (oldCapacity == 0) {
+            newCapacity = DEFAULT_INITIAL_CAPACITY;
+            newThreshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+            table = (Node<K, V> [])new Node[newCapacity];
```

Comment:
if we initialize all necessary parameters in constructor, we wont need this part of code